src/beast/site_scons/site_tools/VSProject.py

```diff
@@ -583,6 +586,43 @@ def _guid(seed, name = None):
     guid = "{%s-%s-%s-%s-%s}" % (d[:8], d[8:12], d[12:16], d[16:20], d[20:32])
     return guid
 
+def _rmExeMod(file_name):
+    if platform.system() != 'Windows':
+        st = os.stat(file_name)
+        os.chmod(file_name, st.st_mode & ~0111)
+        return
+    
+    import win32security
+    import ntsecuritycon
+
+    user, domain, type = win32security.LookupAccountName ("", "Everyone")
+
+    sd = win32security.GetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION)
+    dacl = sd.GetSecurityDescriptorDacl()
+
+    deniedAccess = ntsecuritycon.FILE_EXECUTE
+    dacl.AddAccessDeniedAce(win32security.ACL_REVISION, deniedAccess, user)
+
+    sd.SetSecurityDescriptorDacl(1, dacl, 0)
+    win32security.SetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION, sd)
+
+@contextlib.contextmanager
```

Comment:
Actually - you should probably document that

> bug where I used FILE_EXECUTE_GENERIC instead of FILE_EXECUTE as the flag and VS studio would not read the file and that cost me way too much time.

because if you don't, some poor sap two years from now is going to go through the same exercise.  :-)